kpconv_torch/utils/trainer.py

```diff
@@ -296,23 +285,20 @@ def object_classification_validation(self, net, val_loader):
         :param config: configuration object
         """
 
-        ############
         # Initialize
-        ############
-
         # Choose validation smoothing parameter (0 for no smothing, 0.99 for big smoothing)
         val_smooth = 0.95
 
+        # Number of classes predicted by the model
+        nc_model = val_loader.dataset.num_classes
+
         softmax = torch.nn.Softmax(1)
 
         # Initialize global prediction over all models
         if not hasattr(self, "val_probs"):
-            self.val_probs = np.zeros((val_loader.dataset.num_models, self.num_classes))
+            self.val_probs = np.zeros((val_loader.dataset.num_models, nc_model))
```

Comment:
self.val_probs = np.zeros(
                 (val_loader.dataset.num_models, val_loader.dataset.num_classes)
            )

No need for the extra-variable `nc_model`.

EDIT: Isn't `self.num_classes` still valid?

Reply:
it is

Reply:
Why not to use it?

Reply:
it is done